daemon/container_operations_unix.go

```diff
@@ -177,13 +177,15 @@ func (daemon *Daemon) setupSecretDir(c *container.Container) (setupErr error) {
 			return fmt.Errorf("secret target type is not a file target")
 		}
 
-		targetPath := filepath.Clean(s.File.Name)
-		// ensure that the target is a filename only; no paths allowed
-		if targetPath != filepath.Base(targetPath) {
-			return fmt.Errorf("error creating secret: secret must not be a path")
-		}
+		log := logrus.WithField("name", s.File.Name)
 
-		fPath := filepath.Join(localMountPath, targetPath)
+		// secrets are created in the SecretMountPath at a single level
+		// i.e. /var/run/secrets/foo
```

Comment:
Non-blocking comment nit:  Should this be something like `/var/lib/docker/devicemapper/mnt/<container sha>/secrets/foo` (even though they may not be using devicemapper, just to make it clear that `SecretMountPath` is on the host and not in a container?